File: src/main/java/com/moira/itda/global/entity/CommonCodeDetail.java
package com.moira.itda.global.entity;

import com.moira.itda.domain.common.dto.request.CommonCodeDetailAddRequest;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.ZonedDateTime;

@Builder
@Getter
@Setter
public class CommonCodeDetail {
    private Long id;
    private String key;
    private String korName;
    private String engName;
    private ZonedDateTime createdAt;
    private ZonedDateTime updatedAt;

    public static CommonCodeDetail fromCommonCodeDetailAddRequest(String key, CommonCodeDetailAddRequest request) {
        return CommonCodeDetail.builder()
                .key(key)
                .korName(request.korName())
                .engName(request.engName())
                .createdAt(ZonedDateTime.now())
                .updatedAt(ZonedDateTime.now())
                .build();
    }
}

File: src/main/kotlin/com/moira/itda/global/entity/CommonCodeDetail.kt
package com.moira.itda.global.entity

import com.moira.itda.domain.admin.code.dto.request.AdminCodeDetailAddRequest
import java.time.ZonedDateTime

data class CommonCodeDetail(
    val id: Long?,
    val codeKey: String,
    val engName: String,
    val korName: String,
    val createdAt: ZonedDateTime,
    val updatedAt: ZonedDateTime
) {
    companion object {
        fun fromCommonCodeDetailAddRequest(codeKey: String, request: AdminCodeDetailAddRequest): CommonCodeDetail {
            return CommonCodeDetail(
                id = null,
                codeKey = codeKey,
                engName = request.engName,
                korName = request.korName,
                createdAt = ZonedDateTime.now(),
                updatedAt = ZonedDateTime.now()
            )
        }
    }
}
